File: 0567-permutation-in-string/0567-permutation-in-string.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        vector<int> hash1(26,0);
        vector<int> hash2(26,0);
        if(s1.size()>s2.size())
        return false;

        for(int i=0;i<s1.size();i++){
            hash1[s1[i]-'a']++;
            hash2[s2[i]-'a']++;
        }
        int i=0;
        int j=s1.size();
        
        while(j<s2.size()){
            if(hash1==hash2)
            break;
           
        else{
                
            hash2[s2[i]-'a']--;
            hash2[s2[j]-'a']++;
            i++; j++;
        }
            
        }
        if(hash1==hash2)
        return true;
        return false;
    }
};

File: 0567-permutation-in-string/0567-permutation-in-string.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        if len(s1)>len(s2):
            return False
        hash1=[0]*26
        hash2=[0]*26
        for i in range(len(s1)):
            hash1[ord(s1[i])-ord('a')]+=1
            hash2[ord(s2[i])-ord('a')]+=1
        if hash1==hash2:
            return True
        for i in range(len(s1),len(s2)):
            hash2[ord(s2[i])-ord('a')]+=1
            hash2[ord(s2[i-len(s1)])-ord('a')]-=1
            if hash1==hash2:
                return True
        return False
            
